packages/core/src/browser/shell/application-shell.ts

```diff
@@ -1405,6 +1405,13 @@ export class ApplicationShell extends Widget {
         return [...this.tracker.widgets];
     }
 
+    /**
+     * Returns all tracked dirty widgets.
+     */
+    get dirtyWidgets(): ReadonlyArray<Widget> {
```

Comment:
In general, the `search-in-workspace` should only care about dirty editors, not necessarily all dirty widgets.
Later when calling the method, no filter is performed to extract `Editors`.

Reply:
I think maybe using the `editor-manager.all` method and extracting dirty editors may be sufficient for the implementation of the PR.

Reply:
Just fixed.